AppSearchSample/app/src/main/java/com/android/example/appsearchsample/NoteListItemAdapter.kt

```diff
@@ -88,8 +84,7 @@ class NoteListItemAdapter(private val onDelete: (SearchResult?) -> Unit) :
       }
 
       override fun areContentsTheSame(oldItem: SearchResult, newItem: SearchResult): Boolean {
-        return oldItem.genericDocument.toDocumentClass(Note::class.java) ==
-          newItem.genericDocument.toDocumentClass(Note::class.java)
+        return oldItem == newItem
```

Comment:
This is kind of tricky -- I don't believe the SearchResult class has #equals() overridden. I believe this would then become a referential equality check in Kotlin (checking that the two items are of the same instance) rather than a  structural equality check (checking that the contents of the two items are identical). 

I think it'd be okay to compare the documents (what you had before) and if possible, some matching information thats part of the SearchResult.